experimental/packages/opentelemetry-instrumentation/src/semconvStability.ts

```diff
@@ -106,3 +106,13 @@ export function httpSemconvStabilityFromStr(str: string | undefined) {
 export function databaseSemconvStabilityFromStr(str: string | undefined) {
   return _semconvStabilityFromStr('database', str);
 }
+
+/**
+ * Determine appropriate 'messaging' semconv stability.
+ *
+ * At the time of writing, `messaging` semconv is not yet stabilized. This
+ * function exists in anticipation of that coming.
+ */
+export function messagingSemconvStabilityFromStr(str: string | undefined) {
```

Comment:
I wonder if we should just have `semconvStabilityFromStr()` - and ditch all the others to avoid having to introduce a new function (and having to cut a release) for each semconv category.

Reply:
I could go either way.   The set of functions ("http", "database", "messaging") is tied to the semconv group defining a new scope for the `...OPT_IN` envvar.

One hope, at least for `httpSemconvStabilityFromStr()`, was that we could *remove* the export after the migration period is done. But perhaps that is too hopeful?  With `databaseSemconvStabilityFromStr()` I *could* see 3rd-party instrumentations wanting to use this export, which means we can't really control the approximate 6-month migration window.

Reply:
> I wonder if we should just have semconvStabilityFromStr() - and ditch all the others

I support this. If we want to restrict it we could `export type SemConvArea = "http" | "messaging" | "database";`

Reply:
> If we want to restrict it

Let's not, because then we would be back to requiring a release from core to support a new "area".

I can start a PR.

Reply:
Done in https://github.com/open-telemetry/opentelemetry-js/pull/5684